### PR TITLE
quick fix for truncation warning

### PR DESCRIPTION
**Pull request by alexmontAmazon**
## What does this PR do?
fixes truncation warning when mixing u8s and ints. Technically dangerous if the deviceIndex can exceed 256...

## How was this PR tested?
locally, in the Editor project

### CODE REVIEW

**Review by AMZN-alexpete — APPROVED** [review]

File: Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceDrawPacketBuilder.h [hunk 1]
```diff
@@ -68,7 +68,8 @@ namespace AZ::RHI
 
             for (int deviceIndex = 0; deviceIndex < deviceCount; ++deviceIndex)
             {
-                if (RHI::CheckBit(AZStd::to_underlying(m_deviceMask), deviceIndex))
+                // cast to u8 to prevent warning
```
Comment (moudgils) [inline Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceDrawPacketBuilder.h:71]: Isn't m_deviceMask uint32_t so why doe sit need u8 cast?

**Review by moudgils — APPROVED** [review]

**Comment by lumberyard-employee-dm** [comment]
As a maintainer of SIG-Core, I approve an exception to have this change bypass AR as it is required, to allow other PR builds to succeed.
The AR failure is unrelated to the code changes in the PR
https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE/branches/PR-17326/runs/2/nodes/796/steps/803/log/?start=0

**Comment by AMZN-alexpete** [comment]
I agree, this change is passing all stages and there is some TIAF issue that is unrelated holding it up.

**Comment by jhmueller-huawei** [comment]
Thanks for fixing this so quickly! @moudgils do you think we should add a version of `CheckBit` and similar function, where the parameter can simply be int? Is there a reason for u8? Because even that datatype allows values that are too big (up to 255 instead of 31) and the compilers likely turn that parameter into a simple register anyway that can hold 32 bits (more likely even 64, since there are no 32 bit systems that O3DE supports afaik?).

**Comment by moudgils** [comment]
> Thanks for fixing this so quickly! @moudgils do you think we should add a version of `CheckBit` and similar function, where the parameter can simply be int? Is there a reason for u8? Because even that datatype allows values that are too big (up to 255 instead of 31) and the compilers likely turn that parameter into a simple register anyway that can hold 32 bits (more likely even 64, since there are no 32 bit systems that O3DE supports afaik?).

I would be ok with creating an overloaded CheckBit/SetBit with int param.

**Pull request merged**